gpkit/solution_array.py

```diff
@@ -592,6 +593,25 @@ def savetxt(self, filename="solution.txt", *, printmodel=True, **kwargs):
                 f.write(self.modelstr + "\n")
             f.write(self.table(**kwargs))
 
+    def savejson(self, filename="solution.json"):
```

Comment:
do you want to add an argument to turn only certain variables into json?

Reply:
This would be a great addition. I was talking to Philippe about how it would be nice to have solution jsons organized with a hierarchical structure according to sub-models. It could be nice to have the option to print only variables in a certain sub-model rather than imposing a top down structure on the json.

Reply:
`showvars = self._parse_showvars(showvars)` cleans up some of the various input types, then just intersect / check for inclusion of "k" in there. `parse_showvars` should handle vectorvariables for you, but worth a check if you end up implementing this!

Reply:
Would love to hear any input on what would make sense in the "printing variables of specific sub-models" department

Reply:
with this method you can thankfully just use the models' `varkeys` attribute, as in this section from the performance modeling example:

```python
vars_of_interest = set(AC.varkeys)
# note that there's two ways to access submodels
assert (MISSION["flight segment"]["aircraft performance"]
        is MISSION.fs.aircraftp)
vars_of_interest.update(MISSION.fs.aircraftp.unique_varkeys)
vars_of_interest.add(M["D"])
print(sol.summary(vars_of_interest))
print(sol.table(tables=["loose constraints"]))
```

Reply:
Thanks for the input. Added the showvars parameter to the savejson for applying this logic.